### PR TITLE
BUG: read_excel failing to check older xlrd versions properly 

### LINKED ISSUE #38955: BUG: read_excel() fails when checking __version__ of older xlrd versions
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### When xlrd is installed but is an "older" version (e.g. 1.1.0):

```python
pandas.read_excel("file.xlsx", engine="openpyxl")
```

raises

```python
AttributeError: module 'xlrd' has no attribute '__version__'
```

This specifically occurs in ``pandas/io/excel/_base.py:336``

#### Problem description

This bug occurs regardless of the ``engine`` input because ``pandas.read_excel()`` _always_ checks ``xlrd.__version__`` if any version of xlrd is installed. Older versions of xlrd use ``__VERSION__`` instead of ``__version__`` instead, which results in the AttributeError.

This exception does not occur and everything behaves as expected if:

- xlrd is not installed
- A "newer" version of xlrd is installed (that has the ``__version__`` attribute)

#### Expected Output

According to the documentation, 
```
Changed in version 1.2.0: When engine=None, the following logic will be used to determine the engine:
    
- If path_or_buffer is an OpenDocument format (.odf, .ods, .odt), then odf will be used.
- Otherwise if path_or_buffer is an xls format, xlrd will be used.
- Otherwise if openpyxl is installed, then openpyxl will be used.
- Otherwise if xlrd >= 2.0 is installed, a ValueError will be raised.
- Otherwise xlrd will be used and a FutureWarning will be raised. This case will raise a ValueError in a future version of pandas.

```

Based on this, I believe the intended behavior is:
- read_excel() should _not_ check the xlrd version if engine is ``"openpyxl"``
- If read_excel() does need to know the version of xlrd, it should try both ``__version__`` and ``__VERSION__``

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 3e89b4c4b1580aa890023fc550774e63d499da25
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 3.10.0-1062.4.1.el7.x86_64
Version          : #1 SMP Wed Sep 25 09:42:57 EDT 2019
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US
LOCALE           : en_US.ISO8859-1

pandas           : 1.2.0
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 40.8.0
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : None
xlrd             : 1.1.0
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #38955
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

Probably should be in 1.2.1 given that this occurred after the deprecation of the xlrd engine in 1.2. 
cc @rhshadrach

### CODE REVIEW

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks @lithomas1! Seems to me we should be using `_optional._get_version`. Also I think there  is one in the tests - can you change that too.

**Comment by rhshadrach** [comment]
@lithomas1 I think we still want to use LooseVersion, can apply it to the result of `_get_version`.

File: doc/source/whatsnew/v1.2.2.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 
--
+- Fixed bug in :meth:``read_excel`` that caused it to fail when checking version of older xlrd versions(:issue:`38955`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.2.2.rst:27]: can you move to regression section above

**Comment by simonjayhawkins** [comment]
pre-commit failing

```
pandas/io/excel/_base.py:5:1: F401 'distutils.version.LooseVersion' imported but unused
pandas/tests/io/excel/__init__.py:1:1: F401 'distutils.version.LooseVersion' imported but unused
```

**Comment by jreback** [comment]
@lithomas1 can you merge master and update

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 from pandas._libs.parsers import STR_NA_VALUES
 from pandas._typing import Buffer, DtypeArg, FilePathOrBuffer, StorageOptions
-from pandas.compat._optional import import_optional_dependency
+from pandas.compat._optional import _get_version, import_optional_dependency
```
Comment (jreback) [inline pandas/io/excel/_base.py:18]: we should make this `get_version` and not a private function that is imported across modules. can you make this change as well?

File: doc/source/whatsnew/v1.2.2.rst [hunk 3]
```diff
@@ -14,6 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
+- Fixed bug in :meth:``read_excel`` that caused it to fail when checking version of older xlrd versions(:issue:`38955`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.2.2.rst:17]:
```suggestion
- Fixed regression in :func:`read_excel` that caused it to raise ``AttributeError`` when checking version of older xlrd versions (:issue:`38955`)
```

**Comment by rhshadrach** [comment]
@lithomas1 Can also remove `"_get_version"` from scripts/validate_unwatned_patterns.py

**Review by rhshadrach — APPROVED** [review]
lgtm - thanks @lithomas1!

**Comment by jreback** [comment]
thanks @lithomas1

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Comment by jreback** [comment]
hmm this is not backporting cleanly, @lithomas1 can you follow the instructions above and put up a PR

**Comment by rhshadrach** [comment]
@brobr - can you raise this as a new issue

**Pull request merged**